spec/tiler_spec.coffee

```diff
@@ -1,92 +1,187 @@
+
+      $('#initialize').tiler
+        activeTile: 1
+        isReversible: true
 
     it 'should not include tiles from nested tiler instances', ->
       expect($('#initialize > .tiler-tile').data 'tiler-viewport-id').to.equal 'initialize'
-      expect($('#initialize-nested > .tiler-tile').data 'tiler-viewport-id').to.not.equal 'initialize'
+      expect($('#initialize-nested > .tiler-tile').data 'tiler-viewport-id').to.be_undefined
+
+    it 'should setup enter and exit tiles', ->
+      # tile 1`
+      $activeTile = $('#initialize #tile-initialize-1')
+      expect($activeTile).to.have.class 'tile-initialize-1-animation'
+      expect($activeTile).to.have.class 'active'
+      expect($activeTile).to.have.class 'enter'
+      expect($activeTile).to.have.class 'end'
+      expect($activeTile).to.not.have.class 'previous'
+      expect($activeTile).to.not.have.class 'exit'
+      expect($activeTile).to.not.have.class 'start'
+      expect($activeTile).to.not.have.class 'reverse'
+
+      # tile 2
+      $previousTile = $('#initialize #tile-initialize-2')
+      expect($previousTile).to.have.class 'tile-initialize-1-animation'
+      expect($previousTile).to.have.class 'previous'
+      expect($previousTile).to.have.class 'exit'
+      expect($previousTile).to.have.class 'end'
+      expect($previousTile).to.have.class 'reverse'
+      expect($previousTile).to.not.have.class 'active'
+      expect($previousTile).to.not.have.class 'enter'
+      expect($previousTile).to.not.have.class 'start'
+
+    it 'should match the tiles size to the viewport size', ->
+      expect($('#initialize .tiler-tile').outerWidth()).to.equal 321
+      expect($('#initialize .tiler-tile').outerHeight()).to.equal 123
+
+    it 'should add tile data to the link', ->
+      expect($('button').data('tiler-title')).to.equal 'Tile Initialize 2 Title'
+      expect($('button').data('tiler-animation')).to.equal 'tile-initialize-2-animation'
+
+    context 'with custom activeTile', ->
+      before ->
+        $('#active-tile').tiler
+          activeTile: 2
 
-    describe '_buildLinks', ->
-      it 'should add tile data to the link', ->
-        expect($('button').data('tiler-title')).to.equal 'Tile 1'
-        expect($('button').data('tiler-foo')).to.equal 'Foo 1'
+      it 'should set the active & pervious tiles', ->
+        expect($('#active-tile #tile-active-tile-2')).to.have.class 'active'
+        expect($('#active-tile #tile-active-tile-2')).to.not.have.class 'previous'
+        expect($('#active-tile #tile-active-tile-1')).to.class 'previous'
+        expect($('#active-tile #tile-active-tile-1')).to.not.have.class 'active'
 
-    context 'with options', ->
-      describe 'isReversible', ->
-        before ->
-          $('#reverse-support').tiler
-            isReversible: true
-          $('#reverse-support').tiler('goTo', 2)
-          $('#reverse-support').tiler('goTo', 1)
+  describe 'goTo', ->
+    before ->
+      $('#go-to').tiler()
```

Comment:
there are a lot of selectors i could cache actually... i have a feeling some tests will fail falsely because of it.. but good call.. im going to check it out :)